L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -86,53 +123,310 @@ HGCalConcentratorSuperTriggerCellImpl::getSuperTriggerCellId(int detid) const {
 
       }
 
-      TC_12th =  (rocnum << kRocShift_) | ( (Uprime << kUShift_ | Vprime) & kSplit_v9_ );
+      int TC_split = -1;
 
-      int TC_split =  TC_12th;
-      if (stcSize_.at(thickness) == kSTCsizeCoarse_){
-        TC_split =  rocnum;
+      if ( STCsize > 0 ){
+	TC_split =  (rocnum << kRocShift_) | ( (Uprime << kUShift_ | Vprime) & kSplit_v9_.at( 2 ) );
       }
+      else{
+	
+	if (stcSize_.at(thickness) == kSTCsizeCoarse_){
+	  TC_split =  rocnum;
+	}
+	else{
+	  TC_split =  (rocnum << kRocShift_) | ( (Uprime << kUShift_ | Vprime) & kSplit_v9_.at( stcSize_.at(thickness) ) );
+	}
 
+      }
+    
       return TC_wafer<<kWafer_offset_ | TC_split;
       
     }
     
     
-
   }
   else{
     return -1;
   }
   
 }
 
+void 
+HGCalConcentratorSuperTriggerCellImpl::
+createMissingTriggerCells( std::unordered_map<unsigned,SuperTriggerCell>& STCs, std::vector<l1t::HGCalTriggerCell>& trigCellVecOutput)
+{ 
+
+
+    for (auto& s: STCs){
+
+      bool satisfy = false;
+      //Find and create missing TCs (for super TC 4)
+      int thickness = triggerTools_.thicknessIndex(s.second.GetTCList().at(0),true);
+            
+      if ( stcSize_.at(thickness) ==  4 ){
```

Comment:
All the code below seems overly complicated. I cannot imagine that all this is needed given that the TC->STC is made with only 2 lines of code:
```cpp
TC_split = ( TC_idV8.cell() & kSplit_.at( stcSize_.at(thickness) ) );
TC_wafer<<kWafer_offset_ | TC_split;
```
So doing STC->TC should be just a bit more complicated, building N cell ids from the `TC_split` value.

Reply:
I'll have a think.